File: Latihan2/PHP/Memory.php
<?php
# import class Hardware
include("Hardware.php");

//kelas yang digunakan untuk mengimplementasikan Memory, turunan dari kelas Hardware
class Memory extends Hardware {
    #private attribute dari kelas Memory
    private $frequency = "";
    private $memorySize = 0;
    private $supportsCuda = "";

    #constructor
    public function __construct() {
    }

    #mengeset nilai atribut frequency
    public function setFrequency($frequency) {
        $this->frequency = $frequency;
    }

    #return nilai atribut frequency
    public function getFrequency() {
        return $this->frequency;
    }

    #mengeset nilai atribut memorySize
    public function setMemorySize($memorySize) {
        $this->memorySize = $memorySize;
    }

    #return nilai atribut memorySize
    public function getMemorySize() {
        return $this->memorySize;
    }

    #mengeset nilai atribut supportsCuda
    public function setSupportsCuda($supportsCuda) {
        $this->supportsCuda = $supportsCuda;
    }

    #return nilai atribut supportsCuda
    public function getSupportsCuda() {
        return $this->supportsCuda;
    }

    #menampilkan atribut dari Memory
    public function printMemory() {
        echo "Supports Cuda: " . $this->getSupportsCuda() . "<br/>";
        echo "Memory Size: " . $this->getMemorySize() . "<br/>";
        echo "Frequency: " . $this->getFrequency() . "<br/>";
    }
}

File: Latihan2/Python/Memory.py
#mengimport kelas Hardware
from Hardware import Hardware

#kelas yang digunakan untuk mengimplementasikan Memory, turunan dari kelas Hardware
class Memory(Hardware):
    def __init__(self):
        #constructor
        self.frequency = ""
        self.memorySize = 0
        self.supportsCuda = ""

    def setFrequency(self, frequency):
        #mengeset nilai atribut frequency
        self.frequency = frequency

    def getFrequency(self):
        #return nilai atribut frequency
        return self.frequency

    def setMemorySize(self, memorySize):
        #mengeset nilai atribut memorySize
        self.memorySize = memorySize

    def getMemorySize(self):
        #return nilai atribut memorySize
        return self.memorySize

    def setSupportsCuda(self, supportsCuda):
        #mengeset nilai atribut supportsCuda
        self.supportsCuda = supportsCuda

    def getSupportsCuda(self):
        #return nilai atribut supportsCuda
        return self.supportsCuda

    def printMemory(self):
        #menampilkan atribut dari Memory
        print("Frequency: " + self.getFrequency())
        print("Memory Size: " + str(self.getMemorySize()))
        print("Supports Cuda: " + self.getSupportsCuda())
